src/platform/plugins/shared/data_views/server/index_patterns_api_client.test.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+import { IndexPatternsApiServer } from './index_patterns_api_client';
+import { IndexPatternsFetcher } from './fetcher';
+import { coreMock } from '@kbn/core/server/mocks';
+
+jest.mock('./fetcher');
+
+describe('IndexPatternsApiServer - getFieldsForWildcard', () => {
```

Comment:
adding basic test coverage for the actual problem to fix, there could and should be much more in this area, but where to start? So I decided to focus on the actual issue